### DISCUSSION TITLE
Add a new, sleek, and more usable dock; this idea that Danielle shared is awesome.

### DISCUSSION THREAD

**Discussion by Ditract** [elementary/triage#337] [Ideas]
![Screenshot_20230819-002646](https://github.com/elementary/triage/assets/94490205/90158260-f6f7-404a-b0bb-1092566b9d49)

**Comment by danirabbit** [comment]
Hey thanks for posting this! For some context, here's the Dock project board with results/findings from the user survey we did: https://github.com/orgs/elementary/projects/99

and here's the Dock repo where we've begun writing the new Dock for Wayland: https://github.com/elementary/dock

**Reply by wasantosfi** [reply]
Looking forward to it.

**Reply by doctor-amaton** [reply]
Off topic question but how are you guys working on the Dock? Is the Wayland session stable enough to work on a project that will require wayland to work? Are you using gnome or how? I can't wrap my head around it, I want to help debug and work on it and just installed 7.1 but I can't find any documentation on how you guys are working on this so I can get started

**Reply by danirabbit** [reply]
The Gala dbus methods that the new dock uses work in an X11 session, so you can either use the Pantheon Wayland session or X11 session, you just need to be running Gala and the Dock from git

**Reply by danirabbit** [reply]
Just wanted to post an update that the workspace switcher seems like it’ll be merged very soon! It probably won’t ship for a while until we really get everything ironed out, but it’ll be available to Early Access sponsors for testing and feedback right away when it’s merged: https://github.com/elementary/dock/pull/361

**Comment by kalligator** [comment]
One of the most useful features in plank is the mouse wheel scroll to switch app windows. Please make sure it's included in the new dock! 
In case it's an apps collection folder the wheel could be switching apps instead.

**Comment by Miqueas** [comment]
That looks awesome! Where I can see the original post? :thinking:

**Reply by lenemter** [reply]
https://mastodon.online/@danirabbit/110869040685206007

**Reply by Miqueas** [reply]
Thanks!

**Comment by Romchec** [comment]
Please, just remake plank, but for Wayland. Plank is the only reason why I cant switch to Wayland on my others machines. It's just perfect. There are simply no alternatives to Plank. I don’t think it’s worth overcomplicating it with unnecessary functionality.  The dock should just launch apps and do it well.  Don't worry that it might be similar to what's on Mac OS.  On Linux, there is simply no other dock that is even close to as convenient as Plank, much less for Wayland.  You shouldn't touch too much something that's already working.
I would also like the new dock to have a feature to pin document, images and others files. It's super handy.
And in the end I would like it to have support for old themes.

**Reply by AchillesBoi** [reply]
> There are simply no alternatives to Plank
> On Linux, there is simply no other dock that is even close to as convenient as Plank, much less for Wayland

Dash To Dock on GNOME is far superior to Plank in every way, it's not even up for debate. It works better than Plank on X11 (no low-res or duplicate icons) and it also works on Wayland. The fact that it's only available on GNOME is its only downside, but to say there's no alternative AT ALL on Linux is simply a lie.

**Reply by danirabbit** [reply]
It's not really possible to write a generic dock for any wayland compositor since the extensions that plank would need don't exist in every wayland compositor. We can only write a dock for Gala/Pantheon specifically. This is a good thing because Plank was designed in a time period where the kinds of APIs available to apps were very different so it has some built-in assumptions that don't make sense anymore. This is a time to make changes since we have to do something radically different anyways. We ran a survey a while ago to inform design decisions about the new dock. So things will end up being different from Plank, but that's a good thing :)

**Comment by localdevjs** [comment]
Please consider adding a "ungroup icons" option, which allows us to have one icon in the dock for each window. I understand that some find this behaviour terrible UI, but some of us just work better with a 1 to 1 mapping of icon per window. So much so that that is a major show stopper for me to using plank/elementary in the past. Cinnnamon desktop, KDE, and at least 1 gnome extension do provide this feature as well.

**Comment by kaiserfen** [comment]
One thing I'd love to see in the new dock are these little bullet points, 

A proper way for some form of 3D mode, i.e. what dockey used to have, or at the least a way to move the background of the dock

An easy way for applets to work on it, so people can have any old extension be available for it 

 An application applet on it would be wonderful, as plank had a very outdated one.

**Comment by andres-mestra** [comment]
Greetings, first thank you for the great work you do.
I would like you to add the functionality of having a preview of the windows when you hover over the icon in the dock, this in setting the dock.

**Comment by xvelasqu** [comment]
A feature I love from MacOS is the ability to show folders in the dock:

![dock-stack](https://github.com/elementary/triage/assets/3730793/9155c796-6456-441c-83e3-d82d4771d25f)

The files can be dragged and dropped and they can be displayed according to several settings:

![Screenshot from 2023-11-25 18 48 06](https://github.com/elementary/triage/assets/3730793/e394ea69-8434-4e12-a727-5536efe30589)

And there are different display modes:

![httpatomoreillycomsourceoreillyimages2247512 png](https://github.com/elementary/triage/assets/3730793/62764e81-d001-447c-9fe5-a29b46ca7a9a)

I use it all the time in MacOS. I find it especially useful for the Downloads folder.

**Comment by desiderantes** [comment]
One thing I'd like is a way for apps to offer a menu on counter click when running, instead of just a static list.

**Comment by lennartgoosens** [comment]
@danirabbit In the Mastodon post linked above you wrote that you don't really have a clear idea of how to handle pinned apps yet. I wonder if that's still the case?

Perhaps not even what you meant, but actually, it's always bothered me (a decade ago on MacOS) that the distinction between running apps and pinned apps is so minimal (just the teeny tiny blue arrows, or the dot on MacOS). I'd say the arrows/dots can stay because they also tell you the number of windows per app, but pinned apps imo need to look waaaaay less like running apps than they do now. On rare occassion I accidentally start an app that isn't even running, because I'm trying to bring up it's window just to close it...

But apart from me being blind, more important to me is that pinned but non-running apps draw too much visual attention away from the others (exacerbated by the fact that they occur interspersed, not grouped together by running/pinned status). While having non-running apps so easily on hand is a time saver, having them equally as visible as running apps just reminds me of my distractingly cluttered desk... Which imho doesn't fit well at all with the otherwise extremely clutter-free UX of elementary OS. :)

So, I got inspired by this thread and had some fun with an image editor (and now suddenly it's waaay past bed time... ^^").
**Here is my proposal for handling pinned apps (see attached mock-up):**

![Lennart Goosens   EDIT  8438982601327e4a](https://github.com/elementary/triage/assets/20029818/101c054a-9a38-4f4f-9c6f-af2f0b98999f)

**1. Pinned apps normally recede partly into little shaded pockets. Signalling that they are close on hand, but not in use (just like those plastic cards tucked away in your phone case/wallet). Users can't see the whole icon right away, but 95 % of app's icons will be pretty recognizable even when partly hidden.**
_(I further suggest they are slightly reduced in saturation to avoid drawing attention away from the apps that _are_ currently running.)_

**2. When users mouse-over a pinned app's icon, it temporarily slides out of its pocket for them to see.**
_(If saturation were to be reduced as I suggested, one way of handling the mouse-over effect while still distinguishing it from an already running app, _could_ be to saturate just the center, fading off with a gradient to the sides.)_

**3. Once users start the app, it appears as normal.**
_(Whether or not to keep the pocket shading for a pinned but running app, is a matter of taste. At first I got rid of it, but on second thought I think it discreetly distinguishes pinned apps from non-pinned apps, even when running. "Don't worry, just close me if you like, I'll wait right here in this little pocket for when you need me again.")_

Anyway, this could be one way of doing it. I'd be interested to hear opinions on this! :)
(Also, anyone is free to use/modify/share this idea/mock-up anywhere and for any purpose.)

**Reply by teamcons** [reply]
This is a really cool and lively idea.

**Reply by doctor-amaton** [reply]
This looks pretty cool, if not approved it should still be an option for accessibility

**Reply by kennyb** [reply]
I also like this idea

**Comment by danirabbit** [comment]
I’m going to go ahead and close this as completed since we shipped the new dock in OS 8 and shipped background apps and the workspace switcher in 8.1! If you have more specific ideas about dock please feel free to make new ideas discussions posts or make feature requests directly to the Dock repo
